src/t_zset.c

```diff
@@ -2571,6 +2577,8 @@ void zunionInterDiffGenericCommand(client *c, robj *dstkey, int numkeysIndex, in
     zskiplistNode *znode;
     int withscores = 0;
     unsigned long cardinality = 0;
+    long tmp_limit = 0;      /* In order to do type conversion. */
```

Comment:
Why do we need this temporary variable? It should be enough to make `limit` long.